contracts/protocol/tokenization/base/MintableIncentivizedERC721.sol

```diff
@@ -109,7 +112,28 @@ abstract contract MintableIncentivizedERC721 is
         override
         returns (uint256)
     {
-        return _ERC721Data.userState[account].balance;
+        return
+            _ERC721Data.userState[account].balance +
+            _ERC721Data.userState[account].atomicBalance;
+    }
+
+    function underlyingBalancesOf(address account)
+        public
+        view
+        virtual
+        returns (
+            uint256,
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return (
+            _ERC721Data.userState[account].balance,
+            _ERC721Data.userState[account].atomicBalance,
```

Comment:
we should optimize the gas cost here because we are not using all these anymore in GenericLogic